### PR TITLE
Refactor badge count fetching in Chrome extension

### PR DESCRIPTION
**Pull request by robertknight**
Separate out the logic for querying the server for annotation
counts and processing the response from the logic for updating
the tab state.

 * Replaced code for using XHR and returning the result via a promise
   with the `fetch()` API

 * Move the client code for the Hypothesis /badge endpoint out
   of TabState into its own module and divide up the tests
   accordingly.

 * Ensure the annotation count for a tab is reset to zero if
   querying the /badge endpoint fails.

### CODE REVIEW

File: h/browser/chrome/lib/fetch.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Performs a network fetch for JSON data and returns a Promise
+ * for the result.
+ */
+function fetchJSON(url) {
```
Comment (nickstenning) [inline h/browser/chrome/lib/fetch.js]: You know you're in Chrome here. Can we just use [fetch](https://developer.mozilla.org/en/docs/Web/API/Fetch_API)?

Comment (robertknight) [inline h/browser/chrome/lib/fetch.js]: `fetch()` was made stable in Chrome 42 (April 2015). I figured that's still a little too recent. We could use the 'isomorphic-fetch' polyfill (http://caniuse.com/#feat=fetch) and then just remove that at some point in future.

Comment (robertknight) [inline h/browser/chrome/lib/fetch.js]: Done - Added the 'whatwg-fetch' polyfill (isomorphic-fetch is a wrapper around that, which is also Node-compatible, but we only need the client-side one)

File: h/browser/chrome/lib/tab-state.js [hunk 2]
```diff
@@ -151,31 +148,13 @@ function TabState(initialState, onchange) {
    * @param {string} apiUrl The URL of the Hypothesis API.
    */
   this.updateAnnotationCount = function(tabId, tabUrl, apiUrl) {
```
Comment (nickstenning) [inline h/browser/chrome/lib/tab-state.js]: `apiUrl` is now an unused parameter -- you need to update the function signature and the docstring.

Comment (robertknight) [inline h/browser/chrome/lib/tab-state.js]: Thanks. I will add a lint for this kind of thing in future.

**Comment by nickstenning** [comment]
Otherwise LGTM.

File: h/browser/chrome/lib/tab-state.js (hunk 2 above)
Comment (robertknight) [inline h/browser/chrome/lib/tab-state.js]: Done

**Pull request merged**